ambry-api/src/main/java/com.github.ambry/router/Router.java

```diff
@@ -49,6 +49,15 @@
   public Future<ReadableStreamChannel> getBlob(String blobId);
 
   /**
+   * Requests for blob data asynchronously and returns a future that will eventually contain a
+   * {@link ReadableStreamChannel} that represents blob data on a successful response.
+   * @param blobId The ID of the blob for which blob data is requested.
+   * @param options The options associated with the request.
+   * @return A future that would contain a {@link ReadableStreamChannel} that represents the blob data eventually.
+   */
+  public Future<ReadableStreamChannel> getBlob(String blobId, GetBlobOptions options);
```

Comment:
I don't think we need 4 `getBlob()` methods for every combination of arguments. Is your plan to deprecate and get rid of the ones without `GetBlobOptions` eventually?